pkg/validate/common.go

```diff
@@ -163,7 +162,7 @@ func validateIntRange(path util.Path, val interface{}, min, max int64) util.Erro
 
 // validateCIDR checks whether val is a string with a valid CIDR.
 func validateCIDR(path util.Path, val interface{}) util.Errors {
-	dbgPrintC("validateCIDR (%s): ", val)
+	scope.Debugf("validateCIDR (%s): ", val)
```

Comment:
This is going to mess up the output. dbgPrintC doesn't append a newline at the end. Here and wherever dbgPrintC is used.

Reply:
Then we need to concatenate the debug log and print before return.

Reply:
Yeah, this is exactly why I dragged my feet on moving this to scopes. One possibility is to port to scopes just the files that don't use dbgPrintC. From memory it's just this file that requires it, all others are fair game.

Reply:
You could create a helper function that is called via `defer` which prints logs with optional error, e.g.

```golang
func logWithError(err *error, format string, args ...interface{}) {
	msg := fmt.Sprintf(format, args...)
	if err != nil {
		msg += fmt.Sprintf(": %v", err)
	}
    scope.Debug(msg)
}

func validateCIDR(path util.Path, val interface{}) util.Errors {
    var err error
    defer logWithError(&err, "validateCIRD (%v)", val)
}

```

Reply:
@ostromart @ayj thanks, it helps a lot. We may still need to optimize the logs format in `validateStringList` function, since it coheres with other validation functions.